arvisit/modsen-library-service
# feature/01 implement library service

- Add initial pom.xml
- Add pom.xml
- Add config classes
- Add dtos
- Add main class
- Add controller draft
- Add bootstrap.yml
- Add GitHub Actions
- Add db migration scheme
- Remove request dto for book returning
- Rename records table
- Add mapper dependencies
- Add changelogs for docker context
- Add security client
- Enable feign client
- Add persistence layer
- Add dtos
- Add mappers
- Add util to get user from security context
- Add service layer
- Add validation classes
- Add controller layer
- Add sql-scripts for ITests
- Add configs for ITests
- Add integration tests
- Add dockerfile
- Move dto handling from controller to service
- Add validation if book already exists
- Add mvc tests for addNewBook
- Add tests for getAvailableBooks
- Add mvc tests for getBorrowedBooks
- Correct validation name
- Add mvc tests for borrowBook
- Add mvc tests for returnBook
- Add mvc tests for getBorrowedBookInfo
- Correct check to make possible return book the same day
- Add readme
